Feature/abilities

LukeErlewein opened the pull request:
Adds 4 abilities for both the ranger and the fighter.

DaniereQ commented on src/scripts/playerscripts/FighterPlayer.gd:
```diff
@@ -24,39 +37,157 @@ func _ready() -> void:
+	var explosion_count = 5
+	var delay_between = 2.0
+
+	var explosion_effect_scene := preload("res://src/scenes/gameplayscenes/explosion_effect.tscn")
```
the explosion_effect.tscn is named ExplosionEffect.tscn in the directory. Crashes on game launch

DaniereQ commented on src/scripts/gameplayscripts/Abilities/GravBomb.gd:
```diff
@@ -0,0 +1,23 @@
+class_name GravBomb
+
+func _init(target):
+	cooldown = 7.0
```
This applies to all abilities but `@export` all cooldowns for ease of adjustment.

LukeErlewein replied:
Cooldown is exported in the base skill script

DaniereQ commented on src/scripts/gameplayscripts/Abilities/Grenade.gd:
```diff
@@ -0,0 +1,25 @@
+		var base_transform = Transform2D(facing_dir.angle(), target.global_position)
+		var speed = 100
+		var lifetime = 0.5
+		var damage = 100
```
This goes for all grenade like objects

LukeErlewein replied:
Exported

DaniereQ commented:
We need to talk about whether the dash ability is a short term "sprint" like it currently is or a mono-directional speed boost for a small burst of time similarly to a dodge

DaniereQ commented on src/scripts/playerscripts/FighterPlayer.gd:
```diff
@@ -5,16 +5,29 @@ class_name FighterPlayer extends CharacterBody2D
+@onready var shotgun_ability = $Abilities/ShotgunAbility
+@onready var grenade_ability = $Abilities/GrenadeAbility
+@onready var freeze_grenade_ability = $Abilities/FreezeGrenadeAbility
+@onready var orbital_strike_ability = $Abilities/OrbitalStrikeAbility
```
All of the abilities '@onready' are nonexistent nodes. What is being done to find them?

LukeErlewein replied:
Removed

DaniereQ commented on src/scripts/playerscripts/RangerPlayer.gd:
```diff
@@ -1,51 +1,70 @@
+@onready var pierce_ability = $Abilities/PierceAbility
+@onready var dash_ability = $Abilities/DashAbility
+@onready var grav_grenade_ability = $Abilities/GravGrenadeAbility
+@onready var teleport_ability = $Abilities/TeleportAbility
```
Same is Fighter player. These nodes dont exist?

LukeErlewein replied:
Removed

LukeErlewein commented:
Adressing the following.

Implosion grenade animation isnt in top-down view.
-Found new sprite and used that.

Piercing shot and shotgun blast bullets should be the same size as the default bullets. I could barely tell they were activated.

- Reworked the entire projectile system to allow for different sized projectile sizes.

The ranger teleport should damage all nearby enemies not those unlucky enough to be hit by the small projectiles.
- Made shot of projectiles bigger and added more projectiles

LukeErlewein commented:
Addressing the following

This is more of a design misunderstanding but I thought the Orbital Strike was on mouse location and not a grenade(similar to teleport)
- Changed to be mouse position

Freeze grenade and orbital strike should have lingering visual reminders of the affected location. Once the freeze grenade was thrown I had no idea the affected radius of the ability.

-Added lingering effect.

Abilities shouldnt be cast/start their timers if not unlocked. On game start pressing 4 would start the ability timer and nothing happened because the ability wasnt unlocked. This can be confusing to users as the timer may tell them that something should have happened.

-Disabled ability buttons when the percentage is too low.
